core/src/credential.rs

```diff
@@ -0,0 +1,97 @@
+
+#![allow(unused_variables)]
+#![allow(dead_code)]
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+type VCContext = [String];
+
+pub(crate) const JSON_LD_CONTEXT:  VCContext = ["https://www.w3.org/2018/credentials/v1", "https://www.w3.org/2018/credentials/examples/v1"];
+pub(crate) const CRED_TYPE_PERMANENT_RESIDENT_CARD: str = "PermanentResidentCard";
+pub(crate) const CRED_TYPE_BANK_CARD: str = "BankCard";
+
+type CredentialSubject = serde_json::Value;
+
+enum CredType {
+    PermanentResidentCard(String),
+    BankCard(String)
+}
+
+pub trait VCCredential {
+    fn getCredType (&self) -> String;
+    fn getSubject (&self) -> String;
+    fn getContext (&self) -> [String];
+}
+
+impl VCCredential for VerifiableCredential<'_> {
```

Comment:
This is unnecessary, what we should do is have the trait have the default implementations

Reply:
What I meant is that we should move `create_credential` inside the `DocumentBuilder`. It would live alongside `create_presentation`.

Then `create_credential` can pass everything needed to create a credential to whatever function we define in `credential.rs`. Does that make sense?